### ISSUE TITLE
macOS: Segfault just after connecting from Tracy profiler / capture

### ISSUE THREAD

**Issue by superdump** [wolfpld/tracy#117]
From rust, using tracing-tracy 0.3.0, tracy-client 0.9.0, tracy-client-sys 0.10.0 on macOS, Tracy 0.7.3.

Just after connecting to my instrumented application from the Tracy profiler UI tool or capture CLI tool, my application segfaults. Here is the backtrace:
```
tracy::_memory_allocate(tracy::heap_t*, unsigned long) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/client/tracy_rpmalloc.cpp:0)
tracy::rpmalloc(unsigned long) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/client/tracy_rpmalloc.cpp:2135)
tracy::backtrace_alloc(tracy::backtrace_state*, unsigned long, void (*)(void*, char const*, int), void*) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/libbacktrace/alloc.cpp:62)
tracy::macho_add_dsym(tracy::backtrace_state*, char const*, unsigned long, unsigned char const*, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*)) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/libbacktrace/macho.cpp:878)
tracy::macho_add(tracy::backtrace_state*, char const*, int, long long, unsigned char const*, unsigned long, int, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*), int*) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/libbacktrace/macho.cpp:1161)
tracy::backtrace_initialize(tracy::backtrace_state*, char const*, int, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*)) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/libbacktrace/macho.cpp:1226)
tracy::fileline_initialize(tracy::backtrace_state*, void (*)(void*, char const*, int), void*) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/libbacktrace/fileline.cpp:245)
tracy::backtrace_pcinfo(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/libbacktrace/fileline.cpp:279)
tracy::DecodeCallstackPtr(unsigned long long) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/client/TracyCallstack.cpp:601)
tracy::Profiler::SendCallstackFrame(unsigned long long) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/client/TracyProfiler.cpp:2424)
tracy::Profiler::HandleServerQuery() (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/client/TracyProfiler.cpp:2495)
tracy::Profiler::Worker() (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/client/TracyProfiler.cpp:1537)
tracy::Thread::Launch(void*) (/Users/robswain/.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.10.0/tracy/client/TracyThread.hpp:75)
_pthread_start (@_pthread_start:42)
thread_start (@thread_start:8)
```

**Comment by superdump** [comment]
Also, after this client had segfaulted, I was browsing around the little data that Tracy had obtained, and it too segfaulted. I can make a separate issue for that if it is interesting.

I was sent here from a bug report against tracing-tracy as the segfault is in the tracy client libraries: https://github.com/nagisa/rust_tracy_client/issues/6

**Comment by wolfpld** [comment]
Can you check if the issue is still present, if you `#undef TRACY_ENABLE` in `common/TracyAlloc.hpp`? This would help in narrowing this down to determine if this is ineeded a TLS problem.

Please report the server crash in a separate issue. Can it be be reproduced after saving and reloading the trace from disk?

**Comment by superdump** [comment]
Hmm, I'm not sure how to easily do that if it is to be done in the client. I'll have to have a look at how the tracy-client* rust crates work. @nagisa - any tips?

**Comment by superdump** [comment]
@nagisa I think I figured out how. I checked out https://github.com/nagisa/rust_tracy_client which has tracy as a submodule, and added `#undef TRACY_ENABLE` inside the include guards of `common/TracyAlloc.hpp` but unfortunately linking fails:

```
  = note: Undefined symbols for architecture x86_64:
            "____tracy_emit_messageC", referenced from:
                tracy_client::color_message::h5d07b26a86b21a26 in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.53ama29gppkoep2j.rcgu.o)
            "____tracy_alloc_srcloc_name", referenced from:
                tracy_client::Span::new::h1f501f4fb57d6ece in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.53ama29gppkoep2j.rcgu.o)
            "____tracy_emit_frame_mark", referenced from:
                _$LT$tracing_subscriber..layer..Layered$LT$L$C$S$GT$$u20$as$u20$tracing_core..subscriber..Subscriber$GT$::event::hf063f7a853d11919 in libminkraft-d26891c17746cbbc.rlib(minkraft-d26891c17746cbbc.5cxc3uhyhl03r1e9.rcgu.o)
            "____tracy_emit_zone_begin_alloc", referenced from:
                tracy_client::Span::new::h1f501f4fb57d6ece in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.53ama29gppkoep2j.rcgu.o)
            "____tracy_emit_zone_end", referenced from:
                _$LT$tracy_client..Span$u20$as$u20$core..ops..drop..Drop$GT$::drop::hd27ff8baa4fb40ed in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.48lj0xm9l4vuqppp.rcgu.o)
            "____tracy_emit_message", referenced from:
                tracy_client::message::h6486566b6e17724b in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.53ama29gppkoep2j.rcgu.o)
            "____tracy_emit_zone_begin_alloc_callstack", referenced from:
                tracy_client::Span::new::h1f501f4fb57d6ece in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.53ama29gppkoep2j.rcgu.o)
            "____tracy_init_thread", referenced from:
                tracy_client::Span::new::h1f501f4fb57d6ece in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.53ama29gppkoep2j.rcgu.o)
                tracy_client::color_message::h5d07b26a86b21a26 in libtracy_client-1360db0d3842105b.rlib(tracy_client-1360db0d3842105b.53ama29gppkoep2j.rcgu.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by superdump** [comment]
@nagisa suggested to try the "delayed-init" feature and that got further along, but crashed early on nonetheless. Without the "enable" feature at all, it seems to not show up in the Tracy UI.

**Comment by superdump** [comment]
If I turn off the TRACY_ENABLE feature for the entire client library, does that mean I have to manually implement handling of initialisation of tracy-client somehow?

**Comment by wolfpld** [comment]
If you turn off `TRACY_ENABLE`, then the profiler doesn't exists. Mixing code with and without that define is not and will never be supported. Please only change the one file I have specified.

**Comment by superdump** [comment]
I have done this (and managed to get it to link):

```
diff --git a/common/TracyAlloc.hpp b/common/TracyAlloc.hpp
index 82cc48f2..a24579e1 100644
--- a/common/TracyAlloc.hpp
+++ b/common/TracyAlloc.hpp
@@ -1,6 +1,8 @@
 #ifndef __TRACYALLOC_HPP__
 #define __TRACYALLOC_HPP__
 
+#undef TRACE_ENABLE
+
 #include <stdlib.h>
 
 #ifdef TRACY_ENABLE
```

And unless something weird happened with the build, I get this segfault just after connecting from Tracy profiler to my app:

```
tracy::_memory_allocate(tracy::heap_t*, unsigned long) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/tracy_rpmalloc.cpp:0)
tracy::rpmalloc(unsigned long) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/tracy_rpmalloc.cpp:2135)
tracy::backtrace_alloc(tracy::backtrace_state*, unsigned long, void (*)(void*, char const*, int), void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/alloc.cpp:62)
tracy::macho_add_dsym(tracy::backtrace_state*, char const*, unsigned long, unsigned char const*, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*)) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/macho.cpp:860)
tracy::macho_add(tracy::backtrace_state*, char const*, int, long long, unsigned char const*, unsigned long, int, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*), int*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/macho.cpp:1161)
tracy::backtrace_initialize(tracy::backtrace_state*, char const*, int, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*)) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/macho.cpp:1226)
tracy::fileline_initialize(tracy::backtrace_state*, void (*)(void*, char const*, int), void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/fileline.cpp:245)
tracy::backtrace_pcinfo(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/fileline.cpp:279)
tracy::DecodeCallstackPtr(unsigned long long) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyCallstack.cpp:601)
tracy::Profiler::SendCallstackFrame(unsigned long long) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:2424)
tracy::Profiler::HandleServerQuery() (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:2495)
tracy::Profiler::Worker() (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:1537)
tracy::Thread::Launch(void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyThread.hpp:75)
_pthread_start (@_pthread_start:42)
thread_start (@thread_start:8)
```

**Comment by superdump** [comment]
`alloc.cpp` also includes `../client/tracy_rpmalloc.hpp` and this is the code path where it's segfaulting. `backtrace_alloc` calls `rpmalloc`.

**Comment by wolfpld** [comment]
Yup, there's one non-wrapped allocation in the code: `libbacktrace/alloc.cpp:  ret = rpmalloc (size);`. Will fix that later and let you know.

**Comment by superdump** [comment]
I've swapped out all the rpalloc, rprealloc rpfree, in `libbacktrace/alloc.cpp` and now it doesn't crash.

```
diff --git a/libbacktrace/alloc.cpp b/libbacktrace/alloc.cpp
index d7f02234..0aaf444e 100644
--- a/libbacktrace/alloc.cpp
+++ b/libbacktrace/alloc.cpp
@@ -39,8 +39,6 @@ POSSIBILITY OF SUCH DAMAGE.  */
 #include "backtrace.hpp"
 #include "internal.hpp"
 
-#include "../client/tracy_rpmalloc.hpp"
-
 namespace tracy
 {
 
@@ -59,7 +57,7 @@ backtrace_alloc (struct backtrace_state *state ATTRIBUTE_UNUSED,
 {
   void *ret;
 
-  ret = rpmalloc (size);
+  ret = malloc (size);
   if (ret == NULL)
     {
       if (error_callback)
@@ -76,7 +74,7 @@ backtrace_free (struct backtrace_state *state ATTRIBUTE_UNUSED,
                backtrace_error_callback error_callback ATTRIBUTE_UNUSED,
                void *data ATTRIBUTE_UNUSED)
 {
-  rpfree (p);
+  free(p);
 }
 
 /* Grow VEC by SIZE bytes.  */
@@ -103,7 +101,7 @@ backtrace_vector_grow (struct backtrace_state *state ATTRIBUTE_UNUSED,
       if (alc < vec->size + size)
        alc = vec->size + size;
 
-      base = rprealloc (vec->base, alc);
+      base = realloc (vec->base, alc);
       if (base == NULL)
        {
          error_callback (data, "realloc", errno);
@@ -156,12 +154,12 @@ backtrace_vector_release (struct backtrace_state *state ATTRIBUTE_UNUSED,
     {
       /* As of C17, realloc with size 0 is marked as an obsolescent feature, use
         free instead.  */
-      rpfree (vec->base);
+      free (vec->base);
       vec->base = NULL;
       return 1;
     }
 
-  vec->base = rprealloc (vec->base, vec->size);
+  vec->base = realloc (vec->base, vec->size);
   if (vec->base == NULL)
     {
       error_callback (data, "realloc", errno);
```

**Comment by wolfpld** [comment]
Ok, thanks for checking this out. Will investigate.

**Comment by superdump** [comment]
I spoke too soon:
```
__pthread_kill (@__pthread_kill:6)
pthread_kill (@pthread_kill:109)
abort (@abort:39)
__assert_rtn (@err:3)
unsigned long tracy::moodycamel::ConcurrentQueue<tracy::QueueItem, tracy::moodycamel::ConcurrentQueueDefaultTraits>::ProducerBase::dequeue_bulk<tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3>(tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3) (.cold.5) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/tracy_concurrentqueue.h:2341)
tracy::Profiler::SendSourceLocationPayload(unsigned long long) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:2341)
tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3::operator()(tracy::QueueItem*, unsigned long) const (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:1895)
unsigned long tracy::moodycamel::ConcurrentQueue<tracy::QueueItem, tracy::moodycamel::ConcurrentQueueDefaultTraits>::ExplicitProducer::dequeue_bulk<tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3>(tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/tracy_concurrentqueue.h:1161)
unsigned long tracy::moodycamel::ConcurrentQueue<tracy::QueueItem, tracy::moodycamel::ConcurrentQueueDefaultTraits>::ProducerBase::dequeue_bulk<tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3>(tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/tracy_concurrentqueue.h:942)
unsigned long tracy::moodycamel::ConcurrentQueue<tracy::QueueItem, tracy::moodycamel::ConcurrentQueueDefaultTraits>::try_dequeue_bulk_single<tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3>(tracy::moodycamel::ConsumerToken&, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_2, tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&)::$_3) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/tracy_concurrentqueue.h:574)
tracy::Profiler::Dequeue(tracy::moodycamel::ConsumerToken&) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:1829)
tracy::Profiler::Worker() (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:1501)
tracy::Thread::Launch(void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyThread.hpp:75)
_pthread_start (@_pthread_start:42)
thread_start (@thread_start:8)
```
or
```
tracy::macho_add_symtab(tracy::backtrace_state*, int, unsigned long, int, long long, unsigned int, long long, unsigned int, void (*)(void*, char const*, int), void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/macho.cpp:644)
tracy::macho_add(tracy::backtrace_state*, char const*, int, long long, unsigned char const*, unsigned long, int, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*), int*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/macho.cpp:1099)
tracy::backtrace_initialize(tracy::backtrace_state*, char const*, int, void (*)(void*, char const*, int), void*, int (**)(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*)) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/macho.cpp:1226)
tracy::fileline_initialize(tracy::backtrace_state*, void (*)(void*, char const*, int), void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/fileline.cpp:245)
tracy::backtrace_pcinfo(tracy::backtrace_state*, unsigned long, int (*)(void*, unsigned long, unsigned long, char const*, int, char const*), void (*)(void*, char const*, int), void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/libbacktrace/fileline.cpp:279)
tracy::DecodeCallstackPtr(unsigned long long) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyCallstack.cpp:601)
tracy::Profiler::SendCallstackFrame(unsigned long long) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:2424)
tracy::Profiler::HandleServerQuery() (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:2495)
tracy::Profiler::Worker() (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyProfiler.cpp:1537)
tracy::Thread::Launch(void*) (/Users/robswain/projects/rust/rust_tracy_client/tracy-client-sys/tracy/client/TracyThread.hpp:75)
_pthread_start (@_pthread_start:42)
thread_start (@thread_start:8)
```

**Comment by superdump** [comment]
It seems to be a bit racy if that helps. If I connected sooner rather than waiting for my app to fully start up (not sure if this is relevant as I only tried once so far) then one time I managed to get it to connect and stay connected, gathering a trace, without crashing.

**Comment by wolfpld** [comment]
Are you able to check if you still have these problems when you use other OS, like Linux?

**Comment by superdump** [comment]
If I can get my desktop machine running again then I can try both Windows and Linux. I can't guarantee when I will have time to do that but I'll do my best.

**Comment by wolfpld** [comment]
### First crash log

> dequeue_bulk (...) (tracy-client-sys/tracy/client/tracy_concurrentqueue.h:2341)

What's happening here? `tracy_concurrentqueue.h` only has 1479 lines. Is this some artifact of rust/osx crash stack traces? The next function in the trace is also attributed to line 2341.

The value on which I assume the assert fires up:

```c++
    uint16_t len;
    memcpy( &len, ptr, sizeof( len ) );
    assert( len > 2 );
```

is an user-provided value passed to `___tracy_emit_zone_begin_alloc()`. This value should be obtained using `___tracy_alloc_srcloc()`, which in turn calls `TracyProfiler::AllocSourceLocation()` (snippet simplified to be more clear):

```c++
        const auto sz = uint16_t( 2 + 4 + 4 + functionSz + 1 + sourceSz + 1 + nameSz );
        auto ptr = (char*)tracy_malloc( sz );
        memcpy( ptr, &sz, 2 );
```

Something is happening with this memory between allocation and consumption.

### Second crash log

Hard to tell what's happening there, as it's in the innards of libbacktrace. No such issue was reported before.

Can you run your application through asan/ubsan/valgrind to verify that there's no memory corruption anywhere? Right now everything seems to indicate this being the case here.

**Comment by aclysma** [comment]
Just wanted to mention, I got the same crash on macOS (also using rust) during a capture. I tried running on windows and was able to complete a capture. I'm mostly on macOS though - the reason I'm interested in tracy is for having a UI that works on macOS. :)

**Comment by wolfpld** [comment]
Can you check if this still happens with current master (after 064d2644)?

**Comment by aclysma** [comment]
Yes, this crash still occurs with 0.7.5

**Comment by aclysma** [comment]
Also wanted to add since I just added another issue for ARM/M1, this crash was on intel. (The arm build didn't crash but it doesn't look like it's collecting stack traces properly, which might be what avoids the crash on M1)

**Comment by wolfpld** [comment]
Should be fixed in 8bacc7a9.

**Comment by aclysma** [comment]
The 100% repro I had (stack in the previous screenshot I posted) is resolved on x86 (never happened on ARM) after updating to tracy 0.7.6. Thanks!
